### ISSUE TITLE
Shader Cache in ALBW seems to add more and more entries

### ISSUE THREAD

**Issue by JayFoxRox** [JayFoxRox/citra#44]
Why?

I had 600+ shaders within seconds. Is this normal?

**Comment by JayFoxRox** [comment]
Appears to be normal, however, it still creates a lot of shaders. I guess we should move some variables into uniforms..

**Comment by JayFoxRox** [comment]
One line per shader, abount 1-2 minutes of gameplay (number is random):

```
FFFUUUU! 1804289383
FFFUUUU! 846930886
FFFUUUU! 1681692777
FFFUUUU! 1714636915
FFFUUUU! 1957747793
FFFUUUU! 424238335
FFFUUUU! 719885386
FFFUUUU! 1649760492
FFFUUUU! 596516649
FFFUUUU! 1189641421
FFFUUUU! 1025202362
FFFUUUU! 1350490027
FFFUUUU! 783368690
FFFUUUU! 1102520059
FFFUUUU! 2044897763
FFFUUUU! 1967513926
FFFUUUU! 1365180540
FFFUUUU! 1540383426
FFFUUUU! 304089172
FFFUUUU! 1303455736
FFFUUUU! 35005211
FFFUUUU! 521595368
FFFUUUU! 294702567
FFFUUUU! 1726956429
FFFUUUU! 336465782
FFFUUUU! 861021530
FFFUUUU! 278722862
FFFUUUU! 233665123
FFFUUUU! 2145174067
FFFUUUU! 468703135
FFFUUUU! 1101513929
FFFUUUU! 1801979802
FFFUUUU! 1315634022
FFFUUUU! 635723058
FFFUUUU! 1369133069
FFFUUUU! 1125898167
FFFUUUU! 1059961393
FFFUUUU! 2089018456
FFFUUUU! 628175011
FFFUUUU! 1656478042
FFFUUUU! 1131176229
FFFUUUU! 1653377373
FFFUUUU! 859484421
FFFUUUU! 1914544919
FFFUUUU! 608413784
FFFUUUU! 756898537
FFFUUUU! 1734575198
FFFUUUU! 1973594324
FFFUUUU! 149798315
FFFUUUU! 2038664370
FFFUUUU! 1129566413
FFFUUUU! 184803526
FFFUUUU! 412776091
FFFUUUU! 1424268980
FFFUUUU! 1911759956
FFFUUUU! 749241873
FFFUUUU! 137806862
FFFUUUU! 42999170
FFFUUUU! 982906996
FFFUUUU! 135497281
FFFUUUU! 511702305
FFFUUUU! 2084420925
FFFUUUU! 1937477084
FFFUUUU! 1827336327
FFFUUUU! 572660336
FFFUUUU! 1159126505
FFFUUUU! 805750846
FFFUUUU! 1632621729
FFFUUUU! 1100661313
FFFUUUU! 1433925857
FFFUUUU! 1141616124
FFFUUUU! 84353895
FFFUUUU! 939819582
FFFUUUU! 2001100545
FFFUUUU! 1998898814
FFFUUUU! 1548233367
FFFUUUU! 610515434
FFFUUUU! 1585990364
FFFUUUU! 1374344043
FFFUUUU! 760313750
FFFUUUU! 1477171087
FFFUUUU! 356426808
FFFUUUU! 945117276
FFFUUUU! 1889947178
FFFUUUU! 1780695788
FFFUUUU! 709393584
FFFUUUU! 491705403
FFFUUUU! 1918502651
FFFUUUU! 752392754
FFFUUUU! 1474612399
FFFUUUU! 2053999932
FFFUUUU! 1264095060
FFFUUUU! 1411549676
FFFUUUU! 1843993368
FFFUUUU! 943947739
FFFUUUU! 1984210012
FFFUUUU! 855636226
FFFUUUU! 1749698586
FFFUUUU! 1469348094
FFFUUUU! 1956297539
FFFUUUU! 1036140795
FFFUUUU! 463480570
FFFUUUU! 2040651434
FFFUUUU! 1975960378
FFFUUUU! 317097467
FFFUUUU! 1892066601
FFFUUUU! 1376710097
FFFUUUU! 927612902
FFFUUUU! 1330573317
FFFUUUU! 603570492
FFFUUUU! 1687926652
FFFUUUU! 660260756
FFFUUUU! 959997301
FFFUUUU! 485560280
FFFUUUU! 402724286
FFFUUUU! 593209441
FFFUUUU! 1194953865
FFFUUUU! 894429689
FFFUUUU! 364228444
FFFUUUU! 1947346619
FFFUUUU! 221558440
FFFUUUU! 270744729
FFFUUUU! 1063958031
FFFUUUU! 1633108117
FFFUUUU! 2114738097
FFFUUUU! 2007905771
FFFUUUU! 1469834481
FFFUUUU! 822890675
FFFUUUU! 1610120709
FFFUUUU! 791698927
FFFUUUU! 631704567
FFFUUUU! 498777856
FFFUUUU! 1255179497
FFFUUUU! 524872353
FFFUUUU! 327254586
FFFUUUU! 1572276965
FFFUUUU! 269455306
FFFUUUU! 1703964683
FFFUUUU! 352406219
FFFUUUU! 1600028624
FFFUUUU! 160051528
FFFUUUU! 2040332871
FFFUUUU! 112805732
FFFUUUU! 1120048829
FFFUUUU! 378409503
FFFUUUU! 515530019
FFFUUUU! 1713258270
FFFUUUU! 1573363368
FFFUUUU! 1409959708
FFFUUUU! 2077486715
FFFUUUU! 1373226340
FFFUUUU! 1631518149
FFFUUUU! 200747796
FFFUUUU! 289700723
FFFUUUU! 1117142618
FFFUUUU! 168002245
FFFUUUU! 150122846
FFFUUUU! 439493451
FFFUUUU! 990892921
FFFUUUU! 1760243555
FFFUUUU! 1231192379
FFFUUUU! 1622597488
FFFUUUU! 111537764
FFFUUUU! 338888228
FFFUUUU! 2147469841
FFFUUUU! 438792350
FFFUUUU! 1911165193
FFFUUUU! 269441500
FFFUUUU! 2142757034
FFFUUUU! 116087764
FFFUUUU! 1869470124
FFFUUUU! 155324914
FFFUUUU! 8936987
FFFUUUU! 1982275856
FFFUUUU! 1275373743
FFFUUUU! 387346491
FFFUUUU! 350322227
FFFUUUU! 841148365
FFFUUUU! 1960709859
FFFUUUU! 1760281936
FFFUUUU! 771151432
FFFUUUU! 1186452551
FFFUUUU! 1244316437
FFFUUUU! 971899228
FFFUUUU! 1476153275
FFFUUUU! 213975407
FFFUUUU! 1139901474
FFFUUUU! 1626276121
FFFUUUU! 653468858
FFFUUUU! 2130794395
FFFUUUU! 1239036029
FFFUUUU! 1884661237
FFFUUUU! 1605908235
FFFUUUU! 1350573793
FFFUUUU! 76065818
FFFUUUU! 1605894428
FFFUUUU! 1789366143
FFFUUUU! 1987231011
FFFUUUU! 1875335928
FFFUUUU! 1784639529
FFFUUUU! 2103318776
FFFUUUU! 1597322404
FFFUUUU! 1939964443
FFFUUUU! 2112255763
FFFUUUU! 1432114613
FFFUUUU! 1067854538
FFFUUUU! 352118606
FFFUUUU! 1782436840
FFFUUUU! 1909002904
FFFUUUU! 165344818
FFFUUUU! 1395235128
FFFUUUU! 532670688
FFFUUUU! 1351797369
FFFUUUU! 492067917
FFFUUUU! 1504569917
FFFUUUU! 680466996
FFFUUUU! 706043324
FFFUUUU! 496987743
FFFUUUU! 159259470
FFFUUUU! 1359512183
FFFUUUU! 480298490
FFFUUUU! 1398295499
FFFUUUU! 1096689772
FFFUUUU! 2086206725
FFFUUUU! 601385644
FFFUUUU! 1172755590
FFFUUUU! 1544617505
FFFUUUU! 243268139
FFFUUUU! 1012502954
FFFUUUU! 1272469786
FFFUUUU! 2027907669
FFFUUUU! 968338082
FFFUUUU! 722308542
FFFUUUU! 1820388464
FFFUUUU! 933110197
FFFUUUU! 6939507
FFFUUUU! 740759355
FFFUUUU! 1285228804
FFFUUUU! 1789376348
FFFUUUU! 502278611
FFFUUUU! 1450573622
FFFUUUU! 1037127828
FFFUUUU! 1034949299
FFFUUUU! 654887343
FFFUUUU! 1529195746
FFFUUUU! 392035568
FFFUUUU! 1335354340
FFFUUUU! 87755422
FFFUUUU! 889023311
FFFUUUU! 1494613810
FFFUUUU! 1447267605
FFFUUUU! 1369321801
FFFUUUU! 745425661
FFFUUUU! 396473730
FFFUUUU! 1308044878
FFFUUUU! 1346811305
FFFUUUU! 1569229320
FFFUUUU! 705178736
FFFUUUU! 1590079444
FFFUUUU! 434248626
FFFUUUU! 1977648522
FFFUUUU! 1470503465
FFFUUUU! 1402586708
FFFUUUU! 552473416
FFFUUUU! 1143408282
FFFUUUU! 188213258
FFFUUUU! 559412924
FFFUUUU! 1884167637
FFFUUUU! 1473442062
FFFUUUU! 201305624
FFFUUUU! 238962600
FFFUUUU! 776532036
FFFUUUU! 1238433452
FFFUUUU! 1273911899
FFFUUUU! 1431419379
FFFUUUU! 620145550
FFFUUUU! 1665947468
FFFUUUU! 619290071
FFFUUUU! 707900973
FFFUUUU! 407487131
FFFUUUU! 2113903881
FFFUUUU! 7684930
FFFUUUU! 1776808933
FFFUUUU! 711845894
FFFUUUU! 404158660
FFFUUUU! 937370163
FFFUUUU! 2058657199
FFFUUUU! 1973387981
FFFUUUU! 1642548899
FFFUUUU! 1501252996
FFFUUUU! 260152959
FFFUUUU! 1472713773
FFFUUUU! 824272813
FFFUUUU! 1662739668
FFFUUUU! 2025187190
FFFUUUU! 1967681095
FFFUUUU! 1850952926
FFFUUUU! 437116466
FFFUUUU! 1704365084
FFFUUUU! 1176911340
FFFUUUU! 638422090
FFFUUUU! 1943327684
FFFUUUU! 1953443376
FFFUUUU! 1876855542
FFFUUUU! 1069755936
FFFUUUU! 1237379107
FFFUUUU! 349517445
FFFUUUU! 588219756
FFFUUUU! 1856669179
FFFUUUU! 1057418418
FFFUUUU! 995706887
FFFUUUU! 1823089412
FFFUUUU! 1065103348
FFFUUUU! 625032172
FFFUUUU! 387451659
FFFUUUU! 1469262009
FFFUUUU! 1562402336
FFFUUUU! 298625210
FFFUUUU! 1295166342
FFFUUUU! 1057467587
FFFUUUU! 1799878206
FFFUUUU! 1555319301
FFFUUUU! 382697713
FFFUUUU! 476667372
FFFUUUU! 1070575321
FFFUUUU! 260401255
FFFUUUU! 296864819
FFFUUUU! 774044599
FFFUUUU! 697517721
FFFUUUU! 2001229904
FFFUUUU! 1950955939
FFFUUUU! 1335939811
FFFUUUU! 1797073940
FFFUUUU! 1756915667
FFFUUUU! 1065311705
FFFUUUU! 719346228
FFFUUUU! 846811127
FFFUUUU! 1414829150
FFFUUUU! 1307565984
FFFUUUU! 555996658
FFFUUUU! 324763920
FFFUUUU! 155789224
FFFUUUU! 231602422
FFFUUUU! 1389867269
FFFUUUU! 780821396
FFFUUUU! 619054081
FFFUUUU! 711645630
FFFUUUU! 195740084
FFFUUUU! 917679292
FFFUUUU! 2006811972
FFFUUUU! 1253207672
FFFUUUU! 570073850
FFFUUUU! 1414647625
FFFUUUU! 1635905385
FFFUUUU! 1046741222
FFFUUUU! 337739299
FFFUUUU! 1896306640
FFFUUUU! 1343606042
FFFUUUU! 1111783898
FFFUUUU! 446340713
FFFUUUU! 1197352298
FFFUUUU! 915256190
FFFUUUU! 1782280524
FFFUUUU! 846942590
FFFUUUU! 524688209
FFFUUUU! 700108581
FFFUUUU! 1566288819
FFFUUUU! 1371499336
FFFUUUU! 2114937732
FFFUUUU! 726371155
FFFUUUU! 1927495994
FFFUUUU! 292218004
FFFUUUU! 882160379
FFFUUUU! 11614769
FFFUUUU! 1682085273
FFFUUUU! 1662981776
FFFUUUU! 630668850
FFFUUUU! 246247255
FFFUUUU! 1858721860
FFFUUUU! 1548348142
FFFUUUU! 105575579
FFFUUUU! 964445884
FFFUUUU! 2118421993
FFFUUUU! 1520223205
FFFUUUU! 452867621
FFFUUUU! 1017679567
FFFUUUU! 1857962504
FFFUUUU! 201690613
FFFUUUU! 213801961
FFFUUUU! 822262754
FFFUUUU! 648031326
FFFUUUU! 1411154259
FFFUUUU! 1737518944
FFFUUUU! 282828202
FFFUUUU! 110613202
FFFUUUU! 114723506
FFFUUUU! 982936784
FFFUUUU! 1676902021
FFFUUUU! 1486222842
FFFUUUU! 950390868
FFFUUUU! 255789528
FFFUUUU! 1266235189
FFFUUUU! 1242608872
FFFUUUU! 1137949908
FFFUUUU! 1277849958
FFFUUUU! 777210498
FFFUUUU! 653448036
FFFUUUU! 1908518808
FFFUUUU! 1023457753
FFFUUUU! 364686248
FFFUUUU! 1309383303
FFFUUUU! 1129033333
FFFUUUU! 1329132133
FFFUUUU! 1280321648
FFFUUUU! 501772890
FFFUUUU! 1781999754
FFFUUUU! 150517567
FFFUUUU! 212251746
FFFUUUU! 1983690368
FFFUUUU! 364319529
FFFUUUU! 1034514500
FFFUUUU! 484238046
FFFUUUU! 1775473788
FFFUUUU! 624549797
FFFUUUU! 767066249
FFFUUUU! 1886086990
FFFUUUU! 739273303
FFFUUUU! 1750003033
FFFUUUU! 1415505363
FFFUUUU! 78012497
FFFUUUU! 552910253
FFFUUUU! 1671294892
FFFUUUU! 1344247686
FFFUUUU! 1795519125
FFFUUUU! 661761152
FFFUUUU! 474613996
FFFUUUU! 425245975
FFFUUUU! 1315209188
FFFUUUU! 235649157
FFFUUUU! 1448703729
FFFUUUU! 1679895436
FFFUUUU! 1545032460
FFFUUUU! 430253414
FFFUUUU! 861543921
FFFUUUU! 677870460
FFFUUUU! 932026304
FFFUUUU! 496060028
FFFUUUU! 828388027
FFFUUUU! 1144278050
FFFUUUU! 332266748
FFFUUUU! 1192707556
FFFUUUU! 31308902
FFFUUUU! 816504794
FFFUUUU! 820697697
FFFUUUU! 655858699
FFFUUUU! 1583571043
FFFUUUU! 559301039
FFFUUUU! 1395132002
FFFUUUU! 1186090428
FFFUUUU! 1974806403
FFFUUUU! 1473144500
FFFUUUU! 1739000681
FFFUUUU! 1498617647
FFFUUUU! 669908538
FFFUUUU! 1387036159
FFFUUUU! 12895151
FFFUUUU! 1144522535
FFFUUUU! 1812282134
FFFUUUU! 1328104339
FFFUUUU! 1380171692
FFFUUUU! 1113502215
FFFUUUU! 860516127
FFFUUUU! 777720504
FFFUUUU! 1543755629
FFFUUUU! 1722060049
FFFUUUU! 1455590964
FFFUUUU! 328298285
FFFUUUU! 70636429
FFFUUUU! 136495343
FFFUUUU! 1472576335
FFFUUUU! 402903177
FFFUUUU! 1329202900
FFFUUUU! 1503885238
FFFUUUU! 1219407971
FFFUUUU! 2416949
FFFUUUU! 12260289
FFFUUUU! 655495367
FFFUUUU! 561717988
FFFUUUU! 1407392292
FFFUUUU! 1841585795
FFFUUUU! 389040743
FFFUUUU! 733053144
FFFUUUU! 1433102829
FFFUUUU! 1887658390
FFFUUUU! 1402961682
FFFUUUU! 672655340
FFFUUUU! 1900553541
FFFUUUU! 400000569
FFFUUUU! 337453826
FFFUUUU! 1081174232
FFFUUUU! 1780172261
FFFUUUU! 1450956042
FFFUUUU! 1941690360
FFFUUUU! 410409117
FFFUUUU! 847228023
FFFUUUU! 1516266761
FFFUUUU! 1866000081
FFFUUUU! 1175526309
FFFUUUU! 1586903190
FFFUUUU! 2002495425
FFFUUUU! 500618996
FFFUUUU! 1989806367
FFFUUUU! 1184214677
FFFUUUU! 2004504234
FFFUUUU! 1061730690
FFFUUUU! 1186631626
FFFUUUU! 2016764524
FFFUUUU! 1717226057
FFFUUUU! 1748349614
FFFUUUU! 1276673168
FFFUUUU! 1411328205
FFFUUUU! 2137390358
FFFUUUU! 2009726312
FFFUUUU! 696947386
FFFUUUU! 1877565100
FFFUUUU! 1265204346
FFFUUUU! 1369602726
FFFUUUU! 1630634994
FFFUUUU! 1665204916
FFFUUUU! 1707056552
FFFUUUU! 564325578
FFFUUUU! 1297893529
FFFUUUU! 1010528946
FFFUUUU! 358532290
FFFUUUU! 1708302647
FFFUUUU! 1857756970
FFFUUUU! 1874799051
FFFUUUU! 1426819080
FFFUUUU! 885799631
FFFUUUU! 1314218593
FFFUUUU! 1281830857
FFFUUUU! 1386418627
FFFUUUU! 1156541312
FFFUUUU! 318561886
FFFUUUU! 1243439214
FFFUUUU! 70788355
FFFUUUU! 1505193512
FFFUUUU! 1112720090
FFFUUUU! 1788014412
FFFUUUU! 1106059479
FFFUUUU! 241909610
FFFUUUU! 1051858969
FFFUUUU! 1095966189
FFFUUUU! 104152274
FFFUUUU! 1748806355
FFFUUUU! 826047641
FFFUUUU! 1369356620
FFFUUUU! 970925433
FFFUUUU! 309198987
FFFUUUU! 887077888
FFFUUUU! 530498338
FFFUUUU! 873524566
FFFUUUU! 37487770
FFFUUUU! 1541027284
FFFUUUU! 1232056856
FFFUUUU! 1745790417
FFFUUUU! 1251300606
FFFUUUU! 959372260
FFFUUUU! 1025125849
FFFUUUU! 2137100237
FFFUUUU! 126107205
FFFUUUU! 159473059
FFFUUUU! 1376035217
FFFUUUU! 1282648518
FFFUUUU! 478034945
FFFUUUU! 471990783
FFFUUUU! 1353436873
FFFUUUU! 1983228458
FFFUUUU! 1584710873
FFFUUUU! 993967637
FFFUUUU! 941804289
FFFUUUU! 1826620483
FFFUUUU! 2045826607
FFFUUUU! 2037770478
FFFUUUU! 1930772757
FFFUUUU! 1647149314
FFFUUUU! 716334471
FFFUUUU! 1152645729
FFFUUUU! 470591100
FFFUUUU! 1025533459
FFFUUUU! 2039723618
FFFUUUU! 1001089438
FFFUUUU! 1899058025
FFFUUUU! 2077211388
FFFUUUU! 394633074
FFFUUUU! 983631233
FFFUUUU! 1675518157
FFFUUUU! 1645933681
FFFUUUU! 1943003493
FFFUUUU! 553160358
FFFUUUU! 1635550270
FFFUUUU! 2069110699
FFFUUUU! 712633417
FFFUUUU! 864101839
FFFUUUU! 1204275569
FFFUUUU! 1190668363
FFFUUUU! 1336092622
FFFUUUU! 410228794
FFFUUUU! 1026413173
FFFUUUU! 773319847
FFFUUUU! 1404196431
FFFUUUU! 1968217462
FFFUUUU! 452456682
FFFUUUU! 1302539390
FFFUUUU! 1858504292
FFFUUUU! 235745791
FFFUUUU! 802205057
FFFUUUU! 427355115
FFFUUUU! 1388391521
FFFUUUU! 1272796157
FFFUUUU! 1452888574
FFFUUUU! 1280631491
FFFUUUU! 126401947
FFFUUUU! 1204462951
FFFUUUU! 1210359231
FFFUUUU! 521035021
FFFUUUU! 40610537
FFFUUUU! 738393740
FFFUUUU! 19485054
FFFUUUU! 1983614030
FFFUUUU! 1291554098
FFFUUUU! 1655035325
FFFUUUU! 1905241081
FFFUUUU! 2004187516
FFFUUUU! 371653516
FFFUUUU! 962033002
FFFUUUU! 1047372231
FFFUUUU! 1707746139
FFFUUUU! 1372261796
FFFUUUU! 2073785404
FFFUUUU! 333582338
FFFUUUU! 628974580
FFFUUUU! 1894519218
FFFUUUU! 786039021
FFFUUUU! 1931513970
FFFUUUU! 1605539862
FFFUUUU! 1021784812
FFFUUUU! 586235379
FFFUUUU! 2032894977
FFFUUUU! 262692685
FFFUUUU! 1859031536
FFFUUUU! 1338299904
FFFUUUU! 1543324176
FFFUUUU! 1985433483
```

`wc -l: 654`

**Comment by JayFoxRox** [comment]
Fixed / improved by #46
